public/components/DiscoverAction/GenerateAnomalyDetector.tsx

```diff
@@ -0,0 +1,852 @@
+                })
+                .catch((err: any) => {
+                    dispatch(getDetectorCount(dataSourceId)).then((response: any) => {
+                        const totalDetectors = get(response, 'response.count', 0);
+                        if (totalDetectors === MAX_DETECTORS) {
+                            notifications.toasts.addDanger(
+                                'Cannot create detector - limit of ' +
+                                MAX_DETECTORS +
+                                ' detectors reached'
+                            );
+                        } else {
+                            notifications.toasts.addDanger(
+                                prettifyErrorMessage(
+                                    getErrorMessage(
+                                        err,
+                                        'There was a problem creating the detector'
+                                    )
+                                )
+                            );
+                        }
+                    });
+                });
+            closeFlyout();
+        } catch (e) {
+        } finally {
+            formikProps.setSubmitting(false);
+        }
+    };
+
+    const validateAnomalyDetectorName = async (detectorName: string) => {
+        if (isEmpty(detectorName)) {
+            return 'Detector name cannot be empty';
+        } else {
+            const error = validateDetectorName(detectorName);
+            if (error) {
+                return error;
+            }
+            const resp = await dispatch(matchDetector(detectorName, dataSourceId));
+            const match = get(resp, 'response.match', false);
+            if (!match) {
+                return undefined;
+            }
+            //If more than one detectors found, duplicate exists.
+            if (match) {
+                return 'Duplicate detector name';
+            }
+        }
+    };
+
+    let initialDetectorValue = {
+        name: detectorName,
+        index: [{ label: indexPatternName }],
+        timeField: timeFieldFromIndexPattern,
+        interval: intervalValue,
+        windowDelay: delayValue,
+        shingleSize: DEFAULT_SHINGLE_SIZE,
+        filterQuery: { match_all: {} },
+        description: 'Created based on the OpenSearch Assistant',
+        resultIndex: undefined,
+        filters: [],
+        featureList: [] as FeaturesFormikValues[],
+        categoryFieldEnabled: false,
+        categoryField: [] as string[],
+        realTime: true,
+        historical: false,
+    };
+
+    return (
+        <div className="add-anomaly-detector">
+            <Formik
+                initialValues={initialDetectorValue}
+                onSubmit={handleSubmit}
+                validateOnChange={true}
+                validate={validateFeatures}
+            >
+                {(formikProps) => (
+                    <>
+                        <EuiFlyoutHeader hasBorder>
+                            <EuiTitle>
```

Comment:
Please sync with header redesign team and the UX designer to make sure this is aligned with the header redesign project
Cc @AMoo-Miki @kgcreative

Reply:
Did we have a tech writing review on the naming conventions? It might be a nitpick, but `Suggested anomaly detector` sounds more natural and implies that the detector is being recommended or proposed based on certain criteria, imo